src/fns.js

```diff
@@ -66,12 +69,38 @@ function matchRoute(routes, path) {
         }
       }
 
-      result = { route, params }
-      break
+      results.push({ route, params });
     }
   }
 
-  return result
+  return results
+}
+
+/**
+ * Given a list of matching routes, return a promise which resolves
+ * with the first route in the list that has indicated it should handle
+ * based on its shouldHandle key.
+ * @param routes
+ * @returns {Promise}
+ */
+function filterMatches(routes) {
+  return new Promise((res, rej) => {
+    if (routes.length === 0) {
+      rej('No routes match');
+    } else if (routes.length === 1) {
```

Comment:
do we care about the case if someone accidentally adds a singular route with a shouldHandle fn? i.e. if they thought their route was a match but wasn't, or if they wanted to gate it anyways even if its a singular route. Otherwise this would fail silently and take all users to the route